src/devicecontroller/DeviceControllerFacade.ts

```diff
@@ -0,0 +1,167 @@
+ * called, changes in device availability are broadcast to any registered
+ * [[DeviceChangeObserver]].
+ *
+ * Calling a start function will request permission for the device indicated
+ * by the device id, media stream or track constraint.
+ * For audio input, supply null to will generate a dummy audio stream.
+ * Calling stop to stop the input stream.
+ * Make sure to choose the audio input before joining the session (even if
+ * it is the default device) so that you can offer the user options if the
+ * device cannot be selected before a connection is made.
+ *
+ * Note that in certain situations such as private tabs, the browser may
+ * initially decline to provide device labels for when enumerating devices. If
+ * this is the case, the internal device label trigger function is called to
+ * try to coax the browser in to providing the labels. The default behavior of
+ * the function is to make a microphone and camera access request which, if
+ * successful, will unlock the labels. You may want to override this behavior to
+ * provide a custom UX such as a prompt explaining why microphone and camera
+ * access is being asked for by supplying your own function to
+ * setDeviceLabelTrigger(). To disable the device label trigger, supply a
+ * function that returns a rejected promise instead. For reference, the default
+ * implementation calls getUserMedia for audio and video and returns the promise
+ * to the stream so that the stream can be cleaned up once the labels are
+ * detected.
+ *
+ * ```typescript
+ * (): Promise<MediaStream> => {
+ *   return navigator.mediaDevices.getUserMedia({ audio: true, video: true });
+ * }
+ * ```
+ */
+export default interface DeviceControllerFacade {
+  /**
+   * Lists currently available audio input devices. If `forceUpdate` is set to true, the `deviceInfoCache` will be updated from browser.
+   */
+  listAudioInputDevices(forceUpdate?: boolean): Promise<MediaDeviceInfo[]>;
+
+  /**
+   * Lists currently available video input devices. If `forceUpdate` is set to true, the `deviceInfoCache` will be updated from browser.
+   */
+  listVideoInputDevices(forceUpdate?: boolean): Promise<MediaDeviceInfo[]>;
+
+  /**
+   * Lists currently available audio output devices. If `forceUpdate` is set to true, the `deviceInfoCache` will be updated from browser.
+   */
+  listAudioOutputDevices(forceUpdate?: boolean): Promise<MediaDeviceInfo[]>;
+
+  /**
+   * Selects an audio input device to use. The constraint may be a device id,
+   * `MediaTrackConstraint`, `MediaStream` (containing audio track), or `null` to
+   * indicate no device. It may also be an {@link AudioTransformDevice} to customize the
+   * constraints used or to apply Web Audio transforms.
+   *
+   * The promise will resolve indicating success or it will throw an appropriate error
+   * indicating the failure.
+   */
+  startAudioInput(device: AudioInputDevice): Promise<MediaStream | undefined>;
```

Comment:
Suggest remove `null` from `AudioInputDevice` type and add it in the `startAudioInput` API:
```ts
type AudioInputDevice = Device | AudioTransformDevice;
startAudioInput(device: AudioInputDevice | null): Promise<MediaStream | undefined>;
```

This make `AudioInputDevice` type and `VideoInputDevice` type consistent and easy to tell the difference between `startAudioInput` and `startVideoInput`.
```ts
type VideoInputDevice = Device | VideoTransformDevice;
startVideoInput(device: VideoInputDevice): Promise<MediaStream | undefined>;
```